File: CPP/Program/Car.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once

// Memasukan library yang digunakan, termasuk kelas induk (kelas Vehicle) dari kelas Car
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

// Kelas Car digunakan untuk mempresentasikan data Mobil
// Kelas Car merupakan kelas turunan dari kelas Vehicle dan kelas komposisi dari kelas Garage & Parking Lot
class Car : public Vehicle {
    private:
    // Atribut yang digunakan pada kelas Car terdiri dari jumlah kursi, dan jumlah pintu
        string jumlah_kursi;
        string jumlah_pintu;

    public:
        // Constructor
        Car() {

        }

        // Constructor : Overloading
        Car(string jumlah_kursi, string nama_kendaraan, string plat_nomor, string merk, string tahun_produksi, string warna) : Vehicle(nama_kendaraan, plat_nomor, merk, tahun_produksi, warna) {
            this->jumlah_kursi = jumlah_kursi;
            this->jumlah_pintu = "";
        }


        // Constructor : Overloading
        Car(string jumlah_kursi, string jumlah_pintu, string nama_kendaraan, string plat_nomor, string merk, string tahun_produksi, string warna) : Vehicle(nama_kendaraan, plat_nomor, merk, tahun_produksi, warna) {
            this->jumlah_kursi = jumlah_kursi;
            this->jumlah_pintu = jumlah_pintu;
        }

        // Method set untuk set setiap nilai atribut pada kelas Car
        void setJumlahKursi(string jumlah_kursi) {
            this->jumlah_kursi = jumlah_kursi;
        }

        void setJumlahPintu(string jumlah_pintu) {
            jumlah_pintu = jumlah_pintu;
        }

        // Method get untuk mendapatkan nilai setiap atribut pada kelas Car
        string getJumlahKursi() {
            return jumlah_kursi;
        }

        string getJumlahPintu() {
            return jumlah_pintu;
        }

        // Destructor
        ~Car() {

        }
};

File: Python/Program/Car.py
# Mengimpor kelas Vehicle
from Vehicle import Vehicle

# Kelas Car digunakan untuk mempresentasikan data Mobil
# Kelas Car merupakan kelas turunan dari kelas Vehicle dan kelas komposisi dari kelas Garage & Parking Lot
class Car(Vehicle):
    # Atribut yang digunakan pada kelas Car terdiri dari jumlah kursi, dan jumlah pintu
    # Constructor
    def __init__(self, jumlah_kursi="", jumlah_pintu="", nama_kendaraan="", plat_nomor="", merk="", tahun_produksi="", warna=""):
        super().__init__(nama_kendaraan, plat_nomor, merk, tahun_produksi, warna)
        self.jumlah_kursi = jumlah_kursi
        self.jumlah_pintu = jumlah_pintu

    # Method setter untuk setiap atribut pada kelas Car
    def setJumlahKursi(self, jumlah_kursi):
        self.jumlah_kursi = jumlah_kursi

    def setJumlahPintu(self, jumlah_pintu):
        self.jumlah_pintu = jumlah_pintu

    # Method getter untuk mendapatkan nilai setiap atribut pada kelas Car
    def getJumlahKursi(self):
        return self.jumlah_kursi

    def getJumlahPintu(self):
        return self.jumlah_pintu
